monai/transforms/lazy/functional.py

```diff
@@ -76,30 +79,28 @@ def apply_transforms(
     cumulative_xform = affine_from_pending(pending[0])
     cur_kwargs = kwargs_from_pending(pending[0])
     override_kwargs: dict[str, Any] = {}
-    if mode is not None:
-        override_kwargs[LazyAttr.INTERP_MODE] = mode
-    if padding_mode is not None:
-        override_kwargs[LazyAttr.PADDING_MODE] = padding_mode
-    if align_corners is not None:
-        override_kwargs[LazyAttr.ALIGN_CORNERS] = align_corners
-    if resample_mode is not None:
-        override_kwargs["resample_mode"] = resample_mode
-    override_kwargs[LazyAttr.DTYPE] = data.dtype if dtype is None else dtype
+    if "mode" in overrides:
+        override_kwargs[LazyAttr.INTERP_MODE] = overrides["mode"]
+    if "padding_mode" in overrides:
+        override_kwargs[LazyAttr.PADDING_MODE] = overrides["padding_mode"]
+    if "align_corners" in overrides:
+        override_kwargs[LazyAttr.ALIGN_CORNERS] = overrides["align_corners"]
+    if "resample_mode" in overrides:
+        override_kwargs["resample_mode"] = overrides["resample_mode"]
+    override_kwargs[LazyAttr.DTYPE] = overrides.get("dtype", data.dtype)
 
     for p in pending[1:]:
         new_kwargs = kwargs_from_pending(p)
         if not is_compatible_apply_kwargs(cur_kwargs, new_kwargs):
             # carry out an intermediate resample here due to incompatibility between arguments
             _cur_kwargs = cur_kwargs.copy()
             _cur_kwargs.update(override_kwargs)
-            sp_size = _cur_kwargs.pop(LazyAttr.SHAPE, None)
-            data = resample(data, cumulative_xform, sp_size, _cur_kwargs)
```

Comment:
this sp_size was removed and but in the resample `kwargs.get("spatial_size")` won't get it